src/ray/object_manager/pull_manager.cc

```diff
@@ -123,54 +157,74 @@ void PullManager::UpdatePullsBasedOnAvailableMemory(size_t num_bytes_available)
   }
   num_bytes_available_ = num_bytes_available;
 
-  // While there is available capacity, activate the next pull request.
+  // If there are task requests being pulled and worker requests queued, try to
+  // activate all of the worker requests, even if it puts us over the number of
+  // bytes available. We do this to prioritize worker requests over task
+  // requests.
   std::vector<ObjectID> objects_to_pull;
-  while (num_bytes_being_pulled_ < num_bytes_available_) {
-    // Get the next pull request in the queue.
-    const auto last_request_it = pull_request_bundles_.find(highest_req_id_being_pulled_);
-    auto next_request_it = last_request_it;
-    if (next_request_it == pull_request_bundles_.end()) {
-      // No requests are active. Get the first request in the queue.
-      next_request_it = pull_request_bundles_.begin();
-    } else {
-      next_request_it++;
+  if (highest_task_req_id_being_pulled_ != 0) {
+    bool requests_remaining = true;
+    while (requests_remaining) {
```

Comment:
IIUC this is unconditionally activating all the worker request bundles, whereas in the loop below we still check num_bytes_available_. Is this intended?

It seems like it would be more robust to me to do something like this instead:
```
while any(inactive worker request bundles):
   while not space remaining and any(active task bundles):
      deactivate task bundle
   if space remaining:
      activate worker request bundle
   else:
      break

while space remaining and any(inactive task bundle):
   activate task bundle

while over capacity:
   deactivate task bundle

while over capacity:
  deactivate worker request bundle
```

Reply:
Hmm yeah I think what I had achieves the same thing, but I like yours better!